src/parquet/file/metadata.h

```diff
@@ -0,0 +1,211 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef PARQUET_METADATA_H
+#define PARQUET_METADATA_H
+
+#include <string>
+#include <vector>
+
+#include "parquet/column/properties.h"
+#include "parquet/compression/codec.h"
+#include "parquet/schema/descriptor.h"
+#include "parquet/types.h"
+#include "parquet/util/visibility.h"
+
+namespace parquet {
+
+namespace metadata {
+
+struct ColumnStatistics {
+  int64_t null_count;
+  int64_t distinct_count;
+  const std::string* min;
+  const std::string* max;
+};
+
+class PARQUET_EXPORT SchemaMetaDataReader {
```

Comment:
It also may not be necessary to expose the schema metadata here, since the row group / column chunk / file-level metadata is the only stuff that is currently inaccessible in the public API